g2p/mappings/__init__.py

```diff
@@ -347,7 +347,10 @@ def export_to_dict(self, mapping_type="json", config_only=False):
                 exclude_none=True,
                 exclude={"parent_dir": True},
             )
-        model_dict["rules_path"] = f"{self.in_lang}_to_{self.out_lang}.{mapping_type}"
+        if not model_dict.get("rules_path"):
```

Comment:
should be `if model_dict.get("rules_path") is None` in the off chance that it's a file named `False`? ;-)

Reply:
You've got to be kidding! These file formats can be a nuisance sometimes, but you're right, I've triggered bugs by having `1.0` or `True` as a path elsewhere...
But I also want to fail on "", should the user have typed `rules_path: ` without a value.

So I guess to be safe I'd need to test explicitly for None or "".

Ah-ha, no! By the time we get here, the typing of `rules_path: Optional[Path]` has already been validated by Pydantic and `True` will have triggered a validation error earlier. And an ugly one at that, see #320, which I want to fix.